inc/class-wpseo-meta.php

```diff
@@ -944,39 +946,49 @@ public static function keyword_usage( $keyword, $post_id ) {
 			return [];
 		}
 
-		$query = [
-			'meta_query'     => [
-				'relation' => 'OR',
-				[
-					'key'   => '_yoast_wpseo_focuskw',
-					'value' => $keyword,
-				],
-			],
-			'post__not_in'   => [ $post_id ],
-			'fields'         => 'ids',
-			'post_type'      => 'any',
-
-			/*
-			 * We only need to return zero, one or two results:
-			 * - Zero: keyword hasn't been used before
-			 * - One: Keyword has been used once before
-			 * - Two or more: Keyword has been used twice before
-			 */
-			'posts_per_page' => 2,
-		];
+		/**
+		 * @var Indexable_Repository
+		 */
+		$repository = YoastSEO()->classes->get( Indexable_Repository::class );
+
+		$post_ids = $repository->query()
+			->select( 'object_id' )
+			->where( 'primary_focus_keyword', $keyword )
+			->where( 'object_type', 'post' )
+			->where_not_equal( 'object_id', $post_id )
+			->limit( 2 )
+			->find_array();
+		$post_ids = array_map( function ( $row ) {
+			return (int) $row['object_id'];
+		}, $post_ids );
 
 		// If Yoast SEO Premium is active, get the additional keywords as well.
-		if ( WPSEO_Utils::is_yoast_seo_premium() ) {
-			$query['meta_query'][] = [
-				'key'     => '_yoast_wpseo_focuskeywords',
-				'value'   => sprintf( '"keyword":"%s"', $keyword ),
-				'compare' => 'LIKE',
+		if ( WPSEO_Utils::is_yoast_seo_premium() && count( $post_ids ) < 2 ) {
```

Comment:
Why the check whether `count( $post_ids ) < 2 )`? So if you're in Premium, we only want to have a check on additional keywords when we already found that the primary focus keyword was either not used before, or it was used once before?

Reply:
We only check for the additional keywords if we've not already found two. In that case there's no use for an additional query as we already know the keyword has been used multiple times before.

If we found 0 we want to check as there might be occurrences in the additional keywords.
If we found 1 we want to check as there might be another one in the additional keywords so this would take us from the once tot the multiple error message.
If we already found 2 then no matter how many we find extra the user will get the exact same message.

Reply:
Clear!